src/components/NcModal/NcModal.vue

```diff
@@ -568,6 +576,11 @@ export default {
 		nextButtonAriaLabel() {
 			return t('Next')
 		},
+		dataThemeDarkAttr() {
+			return this.darkHeader
+				? { 'data-theme-dark': true }
```

Comment:
This also needs enforce `light` theme if set, because otherwise you end up with mismatching button colors again